sky/clouds/cloud.py

```diff
@@ -320,14 +330,55 @@ def need_cleanup_after_preemption(self,
         return False
 
     @classmethod
-    def supports(cls,
-                 requested_features: Set[CloudImplementationFeatures]) -> bool:
-        """Returns whether all of the requested features are supported.
+    def check_features_are_supported(
+            cls, requested_features: Set[CloudImplementationFeatures]):
+        """Errors out if the cloud does not support all requested features.
 
         For instance, Lambda Cloud does not support autostop, so
-        Lambda.support({CloudImplementationFeatures.AUTOSTOP}) returns False.
+        Lambda.support({CloudImplementationFeatures.AUTOSTOP}) raises the
+        exception.
+
+        Raises:
+            exceptions.NotSupportedError: If the cloud does not support all the
+            requested features.
         """
-        raise NotImplementedError
+        if cls._CLOUD_IMPLEMENTATION_FEATURES is None:
+            return
+        unsupported_features = (requested_features -
+                                cls._CLOUD_IMPLEMENTATION_FEATURES)
+        if unsupported_features:
+            unsupported_features_str = ', '.join(
+                [f.name for f in unsupported_features])
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.NotSupportedError(
+                    f'Cloud {cls._REPR} does not support the following '
+                    f'features: {unsupported_features_str}')
+
+    @classmethod
+    def check_cluster_name_is_valid(cls, cluster_name: str):
+        """Errors out on invalid cluster names not supported by cloud providers.
+
+        Bans (including but not limited to) names that:
+        - are digits-only
+        - contain underscore (_)
+        Raises:
+            exceptions.InvalidClusterNameError: If the cluster name is invalid.
+        """
+        if cluster_name is None:
+            return
+        valid_regex = '[a-z]([-a-z0-9]*[a-z0-9])?'
+        if re.fullmatch(valid_regex, cluster_name) is None:
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.InvalidClusterNameError(
+                    f'Cluster name "{cluster_name}" is invalid; '
+                    f'ensure it is fully matched by regex: {valid_regex}')
```

Comment:
```suggestion
                    f'ensure it is fully matched by regex (e.g., no underscores or capital letters): {valid_regex}')
```
minor, not feeling strongly: seen such cases

Reply:
Added. Thanks!